zerver/lib/notifications.py

```diff
@@ -113,6 +113,13 @@ def relative_to_full_url(content):
             user_profile.realm.uri + r"/static/generated/emoji/images/emoji/",
             content)
 
+        # Stream links need to be converted from relative to absolute. They
+        # have href values in the form of "/#narrow/stream/...".
+        content = re.sub(
+            r"/#narrow/stream/",
+            user_profile.realm.uri + r"/#narrow/stream/",
+            content)
+
```

Comment:
Can you add a test for this?

Also, I'm confused, since I would have thought that `stream_narrow_url` would take care of this for us; any idea why that isn't working.